x-pack/metricbeat/module/mssql/fetcher.go

```diff
@@ -140,13 +140,7 @@ func (rr *rowsResultHandler) handle(s *s.Schema) (err error) {
 			}
 		}
 
-		result, err := s.Apply(mapOfResults)
-		if err != nil {
-			err = errors.Wrap(err, "error trying to apply schema")
-			logp.Error(err)
-			rr.reporter.Error(err)
-			continue
-		}
+		result, _ := s.Apply(mapOfResults)
```

Comment:
I was wondering if to touch this here (belongs to engine) or in a different PR. The point is that it was generating lot of errors messages that I have seen, were ignored in different modules.

I'm not sure how to proceed here. Comments welcome :)

Reply:
Can you elaborate on the kind of error message? Missing fields? If yes, you can make some fields optional if needed.

Reply:
> The point is that it was generating lot of errors messages that were commonly ignored in different modules.

`schema.Apply` used to report a non-nil error even if no error happened. This made it difficult to handle errors here. I think this was one of the reasons why the error is usually ignored along the code base.

After #7335, `schema.Apply` only reports a non-nil error if an error actually happened (or if a required field is missing), this PR also added:
* Options to specify when a field is required, or if all fields are required or optional.
* A callback for missing fields, so special handling can be implemented for specific cases.
* `KeyError` interface, for errors related to missing or incorrectly formated keys, so these errors can be differentiated for other errors.

>  I remember we came up with a general rule here on what we report and what not.

I wouldn't say that there is a general rule, but with the current implementation is quite possible to make `schema.Apply` to return errors only when needed, by choosing a proper configuration for required and optional fields.

For this particular case, if this module is generating lots of errors, we should look for the source of these errors, e.g:
* If they are caused by expectedly missing fields, we could make these fields optional
* If it is expected that the service returns incorrect or incomplete answers, we should consider this "normal" and don't report the error, but we could log the errors just at the debug level in case we need to investigate something related
* ...

Also regarding reporting errors, I think that this makes sense if the error is caused by some kind of temporal service-related issue, for example if we are being throttled by an overloaded service.

Reply:
@sayden Based on the above comment from Jaime we should handle the error here. Can you share details on the error?

Reply:
This doesn't apply anymore by using Optional fields and generating a single event  (instead of an event for each field). For reference, I'll link this conversation in the SQL helper issue [here](https://github.com/elastic/beats/issues/9416)

Reply:
This means this change should be reverted I would assume?

Reply:
Means that this entire file will be removed soon, as it works by using deep abstractions that make it look closer to a _SQL helper_ than the needs we have for this particular module.

Reply:
Do we need this change for this PR? If not, lets revert it.